src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs

```diff
@@ -779,6 +779,25 @@ Point getChildLocation()
 				}
 
 				var comboRect = combo.GetAbsoluteBoundsRect();
+
+#if __IOS__
+				// We might get to a situation where the calculations performed during
+				// .GetAbsoluteBoundsRect() did not return the correct value
```

Comment:
What about other callers of `GetAbsoluteBoundsRect`? Do they have the same issue? In this case should the fix go into `GetAbsoluteBoundsRect` itself?

Reply:
I think we need to understand the root cause of `GetAbsoluteBoundsRect` returning an incorrect value, in case the issue is deeper somewhere.

Reply:
It could be that GetAbsoluteBoundsRect is not able to determine how to get out of its own tree ancestors, and determine its own anchor.

Reply:
The reason I know it:

`GetAbsoluteBoundsRect ()` uses the `Windows.Current.Content `as the `EndPoint` when calculating the position of a Children View. This means it will start iterating from the View (in this case, the ComboBox) til it finds the `To` which will almost always be the Frame. 

The way we launch a Modal Page is by mimic-ing a Multi-Frame navigation.

A `ComboBox` hierarchy inside the Main Window looks like this

```xml
><Uno_UI_Controls_Window: 0x7f7f8d91cf50
  ><UITransitionView: 0x7f7f8cf0a820
    ><UIDropShadowView: 0x7f7f8cf0d0d0
      ><UILayoutContainerView: 0x7f7f8d9206d0
        ><UINavigationTransitionView: 0x7f7f8da781b0
        >Uno.UI.Xaml.Core.RootVisual-(393x852)@(0,0)
          >Windows.UI.Xaml.Controls.Border-(393x852)@(
            >Windows.UI.Xaml.Controls.Frame-(393x852)@(0,0)
              >Uno.Toolkit.UI.NativeFramePresenter-(393x852)@(0,0)
                ><UILayoutContainerView: 0x7f7f8cf229f0
                  ><UINavigationTransitionView: 0x7f7f8cf26ee0
                    ><UIViewControllerWrapperView: 0x7f7f8cf3c450
                      >SampleComboBoxFlyout.MainPage-(393x852)@(0,0)
                        >SampleComboBoxFlyout.MultiFrame-'NavigationRoot'-(393x852)@(0,0)
                          >Windows.UI.Xaml.Controls.Grid-(393x852)@(0,0)
                            >Windows.UI.Xaml.Controls.ComboBox-(123x32)
```

Where it will be part of the `Windows.UI.Xaml.Controls.Frame` 

When we put a `ComboBox` inside a Modal Page, the hierarchy looks like this now

```xml
><Uno_UI_Controls_Window: 0x7f7f8d91cf50
  ><UITransitionView: 0x7f7f8d818d50
    ><UIDimmingView: 0x7f7f8d822e70
    ><UIDropShadowView: 0x7f7f8d825550
      ><_UIRoundedRectShadowView: 0x7f7f8d8257b0
      ><UIView: 0x7f7f8d825a00
        ><UIView: 0x7f7f8d81fbb0
          *>SampleComboBoxFlyout.ModalPage-(393x783)@(0,0)
            >Windows.UI.Xaml.Controls.Grid-(393x783)@(0,0)
              >Windows.UI.Xaml.Controls.ComboBox-(0x0)@(0,0)
```

As can be seen, this hierarchy does not contains a `Frame` but we are telling that we want to iterate all the way up til we reach a `Frame`. This causes `GetAbsoluteBoundsRect ` to return invalid information for the Frame.

As the comment states, this workaround is a way to ensure that if we cannot get the right location from `GetAbsoluteBoundsRect ` we at least use the location information from the ComboBox itself.

Modifying the `ComboBox` sounds like the less intrusive solution we can have, taking into account that this is an edge case.

[Related method](https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/UIElement.iOS.cs#L164) found on UIElement.iOS.cs

Hope this helps clear any doubts.

cc: @jeromelaban  @Youssef1313

Reply:
Btw: Also tried using a `Frame` as part of the Modal Page where the ComboBox is displayed. Unfortunately, this breaks the whole app since we are not expecting to find more than one `Frame` in the tree hierarchy.